mingw-w64-cross-winpthreads/PKGBUILD

```diff
@@ -43,23 +39,60 @@ prepare() {
   autoreconf -vfi
 }
 
+_build() {
+  _target=$1
+  mkdir -p ${srcdir}/winpthreads-build-${_target} && cd ${srcdir}/winpthreads-build-${_target}
+  unset CC
+  ${srcdir}/mingw-w64/mingw-w64-libraries/winpthreads/configure \
+    --prefix=/opt/${_target} \
+    --build=${CHOST} \
+    --host=${_target} \
+    --enable-static \
+    --enable-shared
+  make
+}
+
+_package() {
+  _target=$1
+  cd ${srcdir}/winpthreads-build-${_target}
+  make DESTDIR=${pkgdir} install
+}
+
 build() {
-  for _target in ${_targets}; do
-    mkdir -p ${srcdir}/winpthreads-build-${_target} && cd ${srcdir}/winpthreads-build-${_target}
-    unset CC
-    ${srcdir}/mingw-w64/mingw-w64-libraries/winpthreads/configure \
-      --prefix=/opt/${_target} \
-      --build=${CHOST} \
-      --host=${_target} \
-      --enable-static \
-      --enable-shared
-    make
-  done
+  _build "x86_64-w64-mingw32ucrt"
+  _build "i686-w64-mingw32"
+  _build "x86_64-w64-mingw32"
+}
+
+package_mingw-w64-cross-winpthreads() {
+  pkgdesc+=" (meta package, for backward compatibility)"
+  depends=(
+    "${_mingw_suff}-ucrt64-${_realname}"
+    "${_mingw_suff}-mingw32-${_realname}"
+    "${_mingw_suff}-mingw64-${_realname}"
+  )
+  provides=("${_mingw_suff}-${_realname}-git")
+  conflicts=("${_mingw_suff}-${_realname}-git")
+  replaces=("${_mingw_suff}-${_realname}-git")
 }
 
-package() {
-  for _target in ${_targets}; do
-    cd ${srcdir}/winpthreads-build-${_target}
-    make DESTDIR=${pkgdir} install
-  done
+package_mingw-w64-cross-ucrt64-winpthreads() {
```

Comment:
Is there a possibility to build only one variant of the package? If all variants need to be build at once, it will require porting/bootstrapping all those variants for aarch64...

Reply:
I could check add a check in _build() for which package is defined in the pkgname array. Then it would only build/package what is defined in pkgname. Would that help?

Reply:
Yes, it would make the necessary changes minimal.

Reply:
You could also have your own PKGBUILD building just aarch64

Reply:
btw, after this is all done I would look into integrating https://github.com/Windows-on-ARM-Experiments/MSYS2-packages/commits/woarm64/

Reply:
Thank you. Any help would be appreciated.